cockpit_api/src/test/java/com/cockpit/api/service/jiragateway/JiraGatewayServiceTest.java

```diff
@@ -69,4 +98,75 @@ public void whenUpdateTotalNbUserStoriesTaskRunsThenUpdateTotalNbUserStoriesInSp
         Assert.assertEquals(5, mockSprint.getTotalNbUs().intValue());
     }
 
+    @Test
+    public void whenDeleteOnJiraThenDeleteOnCockpit() throws UnirestException, JsonProcessingException, JSONException {
+        Jira mockJira = new Jira();
+        Sprint mockSprint = new Sprint();
+
+        mockJira.setId(99999L);
+        mockJira.setJiraProjectKey("TEST");
+
+        mockSprint.setJira(mockJira);
+        mockSprint.setId(99999L);
+        mockSprint.setJiraSprintId(10024);
+        mockSprint.setSprintNumber(2);
+
+        UserStory mockUserStory = new UserStory();
+
+        mockUserStory.setJira(mockJira);
+        mockUserStory.setId(99999L);
+        mockUserStory.setSprint(mockSprint);
+        mockUserStory.setIssueKey("TC");
+
+
+        List<UserStory> mockUserStoryList = new ArrayList<>();
+        mockUserStoryList.add(mockUserStory);
+        mockUserStoryList.add(mockUserStory);
+
+        List<Sprint> mockSprintList = new ArrayList<>();
+        mockSprintList.add(mockSprint);
+        mockSprintList.add(mockSprint);
+
+        List<Jira> mockJiraList = new ArrayList<>();
+        mockJiraList.add(mockJira);
+        mockJiraList.add(mockJira);
+
+        Mockito.when(jiraRepository.findAllByOrderById()).thenReturn(mockJiraList);
+        JSONArray mockJiraJsonArray = new JSONArray();
+        for (Jira jira : mockJiraList) {
+            mockJiraJsonArray.put(jira);
+        }
+
+        JiraGatewayService mockJiraGatewayService = mock(JiraGatewayService.class);
+        ReflectionTestUtils.setField(jiraGatewayService, "username", username);
+        ReflectionTestUtils.setField(jiraGatewayService, "token", token);
+        ReflectionTestUtils.setField(jiraGatewayService, "jiraUrl", jiraUrl);
+
+        Mockito.when(mockJiraGatewayService.getJiraProjects()).thenReturn(mockJiraJsonArray);
+        jiraGatewayService.deleteJiraProjects();
+
+        Mockito.when(sprintRepository.findAll()).thenReturn(mockSprintList);
+        JSONArray mockSprintJsonArray = new JSONArray();
+        for (Sprint sprint : mockSprintList) {
+            mockSprintJsonArray.put(sprint);
+        }
+        Mockito.when(mockJiraGatewayService.getJiraSprints(anyLong())).thenReturn(mockSprintJsonArray);
+        jiraGatewayService.deleteJiraSprint();
+
+        Mockito.when(userStoryRepository.findAll()).thenReturn(mockUserStoryList);
+        JSONArray mockUserStoryJsonArray = new JSONArray();
+        for (UserStory userStory : mockUserStoryList) {
+            mockUserStoryJsonArray.put(userStory);
+        }
+        Mockito.when(mockJiraGatewayService.getJiraIssueList()).thenReturn(mockUserStoryJsonArray);
+        jiraGatewayService.deleteJiraIssues();
+
+        assertThat(userStoryRepository.count()).isEqualTo(0);
```

Comment:
These 3 values are always 0 even before the jiraGatewayService.deleteJiraIssues(), is it normal ?

Reply:
Currently fixing